go/vt/wrangler/traffic_switcher.go

```diff
@@ -1910,6 +1964,274 @@ func (ts *trafficSwitcher) isSequenceParticipating(ctx context.Context) (bool, e
+		}
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, false, err
+	}
+
+	return sequencesByBackingTable, allFullyQualified, nil
+}
+
+// initializeTargetSequences initializes the backing sequence tables
+// using a map keyed by the backing sequence table name.
+//
+// The backing tables must have already been created. This function will
+// then ensure that the next value is set to a value greater than any
+// currently stored in the using table on the target keyspace. If the
+// backing table is updated to a new higher value then it will also tell
+// the primary tablet serving the sequence to refresh/reset its cache to
+// be sure that it does not provide a value that is less than the current max.
+func (ts *trafficSwitcher) initializeTargetSequences(ctx context.Context, sequencesByBackingTable map[string]*sequenceMetadata) error {
+	initSequenceTable := func(ictx context.Context, sequenceTableName string, sequenceMetadata *sequenceMetadata) error {
+		// Now we need to run this query on the target shards in order
+		// to get the max value and set the next id for the sequence to
+		// a higher value.
+		shardResults := make([]int64, 0, len(ts.TargetShards()))
+		srMu := sync.Mutex{}
+		ierr := ts.ForAllTargets(func(target *workflow.MigrationTarget) error {
+			primary := target.GetPrimary()
+			if primary == nil || primary.GetAlias() == nil {
+				return vterrors.Errorf(vtrpcpb.Code_INTERNAL, "no primary tablet found for target shard %s/%s",
+					ts.targetKeyspace, target.GetShard().ShardName())
+			}
+			query := sqlparser.BuildParsedQuery(sqlGetMaxSequenceVal,
+				sqlescape.EscapeID(sequenceMetadata.usingTableDefinition.AutoIncrement.Column),
+				sqlescape.EscapeID(sequenceMetadata.usingTableDBName),
+				sqlescape.EscapeID(sequenceMetadata.usingTableName),
+			)
+			qr, terr := ts.wr.ExecuteFetchAsApp(ictx, primary.GetAlias(), true, query.Query, 1)
+			if terr != nil || len(qr.Rows) != 1 {
+				return vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get the max used sequence value for target table %s.%s in order to initialize the backing sequence table: %v",
+					ts.targetKeyspace, sequenceMetadata.usingTableName, terr)
+			}
+			maxID, terr := sqltypes.Proto3ToResult(qr).Rows[0][0].ToInt64()
+			if terr != nil {
+				return vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get the max used sequence value for target table %s.%s in order to initialize the backing sequence table: %v",
+					ts.targetKeyspace, sequenceMetadata.usingTableName, terr)
+			}
+			srMu.Lock()
+			defer srMu.Unlock()
+			shardResults = append(shardResults, maxID)
+			return nil
+		})
+		if ierr != nil {
+			return ierr
+		}
+		select {
+		case <-ictx.Done():
+			return ictx.Err()
+		default:
+		}
+		// Sort the values to find the max value across all shards.
+		sort.Slice(shardResults, func(i, j int) bool {
+			return shardResults[i] < shardResults[j]
+		})
+		nextVal := shardResults[len(shardResults)-1] + 1
```

Comment:
I think you want a bounds check in case shardResults is empty or else you'll access the -1th index here and it'll panic.